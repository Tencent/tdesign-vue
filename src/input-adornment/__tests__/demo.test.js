/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import selectVue from '@/src/input-adornment/_example/select.vue';
import textVue from '@/src/input-adornment/_example/text.vue';

const mapper = {
  selectVue,
  textVue,
};

describe('InputAdornment', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`InputAdornment ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
