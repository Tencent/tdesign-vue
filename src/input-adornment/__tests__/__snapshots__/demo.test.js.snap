// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAdornment InputAdornment selectVue demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
  style="max-width: 500px;"
>
  <div
    class="t-input-adornment t-input-adornment--prepend"
  >
    <span
      class="t-input-adornment__prepend"
    >
      <div
        class="t-select__wrap"
      >
        <div
          class="t-input__wrap t-select-input--borderless t-select"
        >
          <div
            class="t-input t-size-m t-is-default t-is-readonly t-input--auto-width t-input--prefix t-input--suffix"
          >
            <div
              class="t-input__prefix"
            />
            <input
              autocomplete=""
              class="t-input__inner"
              placeholder="请选择"
              readonly="readonly"
              type="text"
              unselectable="on"
            />
            <span
              class="t-input__input-pre"
            >
              http://
            </span>
            <span
              class="t-input__suffix t-input__suffix-icon"
            >
              <svg
                class="t-fake-arrow t-select__right-icon"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                  stroke="black"
                  stroke-opacity="0.9"
                  stroke-width="1.3"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </span>
    <div
      class="t-input__wrap"
    >
      <div
        class="t-input t-size-m t-is-default"
      >
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入内容"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
  </div>
   
  <div
    class="t-input-adornment t-input-adornment--append"
  >
    <div
      class="t-input__wrap t-tag-input t-tag-input--break-line"
    >
      <div
        class="t-input t-size-m t-is-default t-input--prefix"
      >
        <div
          class="t-input__prefix"
        />
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入内容"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
    <span
      class="t-input-adornment__append"
    >
      <div
        class="t-select__wrap"
      >
        <div
          class="t-input__wrap t-select-input--borderless t-select"
        >
          <div
            class="t-input t-size-m t-is-default t-is-readonly t-input--auto-width t-input--prefix t-input--suffix"
          >
            <div
              class="t-input__prefix"
            />
            <input
              autocomplete=""
              class="t-input__inner"
              placeholder="请选择"
              readonly="readonly"
              type="text"
              unselectable="on"
            />
            <span
              class="t-input__input-pre"
            >
              .cn
            </span>
            <span
              class="t-input__suffix t-input__suffix-icon"
            >
              <svg
                class="t-fake-arrow t-select__right-icon"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                  stroke="black"
                  stroke-opacity="0.9"
                  stroke-width="1.3"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </span>
  </div>
   
  <div
    class="t-input-adornment t-input-adornment--prepend t-input-adornment--append"
  >
    <span
      class="t-input-adornment__prepend"
    >
      <div
        class="t-select__wrap"
      >
        <div
          class="t-input__wrap t-select-input--borderless t-select"
        >
          <div
            class="t-input t-size-m t-is-default t-is-readonly t-input--auto-width t-input--prefix t-input--suffix"
          >
            <div
              class="t-input__prefix"
            />
            <input
              autocomplete=""
              class="t-input__inner"
              placeholder="请选择"
              readonly="readonly"
              type="text"
              unselectable="on"
            />
            <span
              class="t-input__input-pre"
            >
              http://
            </span>
            <span
              class="t-input__suffix t-input__suffix-icon"
            >
              <svg
                class="t-fake-arrow t-select__right-icon"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                  stroke="black"
                  stroke-opacity="0.9"
                  stroke-width="1.3"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </span>
    <div
      class="t-select__wrap"
    >
      <div
        class="t-input__wrap t-select"
      >
        <div
          class="t-input t-size-m t-is-default t-is-readonly t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请选择"
            readonly="readonly"
            type="text"
            unselectable="on"
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-fake-arrow t-select__right-icon"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <span
      class="t-input-adornment__append"
    >
      <div
        class="t-select__wrap"
      >
        <div
          class="t-input__wrap t-select-input--borderless t-select"
        >
          <div
            class="t-input t-size-m t-is-default t-is-readonly t-input--auto-width t-input--prefix t-input--suffix"
          >
            <div
              class="t-input__prefix"
            />
            <input
              autocomplete=""
              class="t-input__inner"
              placeholder="请选择"
              readonly="readonly"
              type="text"
              unselectable="on"
            />
            <span
              class="t-input__input-pre"
            >
              .cn
            </span>
            <span
              class="t-input__suffix t-input__suffix-icon"
            >
              <svg
                class="t-fake-arrow t-select__right-icon"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                  stroke="black"
                  stroke-opacity="0.9"
                  stroke-width="1.3"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`InputAdornment InputAdornment textVue demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
  style="max-width: 500px;"
>
  <div
    class="t-input-adornment t-input-adornment--prepend"
  >
    <span
      class="t-input-adornment__prepend"
    >
      http://
    </span>
    <div
      class="t-input__wrap"
    >
      <div
        class="t-input t-size-m t-is-default"
      >
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入内容"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
  </div>
   
  <div
    class="t-input-adornment t-input-adornment--append"
  >
    <div
      class="t-input__wrap t-tag-input t-tag-input--break-line"
    >
      <div
        class="t-input t-size-m t-is-default t-input--prefix"
      >
        <div
          class="t-input__prefix"
        />
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入内容"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
    <span
      class="t-input-adornment__append"
    >
      .com
    </span>
  </div>
   
  <div
    class="t-input-adornment t-input-adornment--prepend t-input-adornment--append"
  >
    <span
      class="t-input-adornment__prepend"
    >
      http://
    </span>
    <div
      class="t-input__wrap"
    >
      <div
        class="t-input t-size-m t-is-default"
      >
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入内容"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
    <span
      class="t-input-adornment__append"
    >
      .com
    </span>
  </div>
</div>
`;
